packages/plugin-ext/src/main/browser/plugin-icon-theme-service.ts

```diff
@@ -127,7 +128,7 @@ export class PluginIconTheme extends PluginIconThemeDefinition implements IconTh
     @postConstruct()
     protected init(): void {
         Object.assign(this, this.definition);
-        this.packageUri = new URI(this.packagePath);
+        this.packageUri = URI.file(this.packagePath);
```

Comment:
It cannot be used here: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#frontend-fs-path

Reply:
It wouldn't for me because the `:` and `\` would mess up the Uri and I couldn't get `relative` to work... I'll try again with path's and discard the `static file` function entirely.

Reply:
So the packageUri looks like: `C:\...\...` while the normalized icon uri path looks like: `/c:/.../...`.
To use the relative function the packageUri path also has to be in the form: `/c:/.../...` with only forwardslashes and one forwardslash at the beginning. But by constructing a `URI` object with the packagePath the path will result in: `/User/...`. I could change the [normalizeDrive](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/path.ts) function to automatically insert the missing `/` at the beginning. But then I'd have to use the Path class inside of [plugin-icon-theme-service.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/plugin-icon-theme-service.ts) or I could change the URI constructor in [uri.ts](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/uri.ts) to use the string ( if one is given )  and first construct a Path that then is automatically corrected with the beginning `/` and use the resulting string to parse with the `Uri.parse` function. I'll update my repo and then you can decide to do that or to just do it in the [plugin-icon-theme-service.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/plugin-icon-theme-service.ts) and use `Path` there.

Reply:
We should avoid changing core types as URI and Path.

Reply:
For me it doesn't even work in the browser. The packagePath looks like: `C:\...\...` for me and after the new URI was created the normalized path of that Uri will result in: `\Users\luca\Desktop\tmp\theia\plugins\vscode-builtin-icon-theme-seti\extension\` while the normalized path of the icon uri is: `/c:/Users/luca/Desktop/tmp/theia/plugins/vscode-builtin-icon-theme-seti/extension/icons/seti.woff`. To get the relative path both must start the same way though: 
```
    relative(path: Path): Path | undefined {
        if (this.raw === path.raw) {
            return new Path('');
        }
        if (!this.raw || !path.raw) {
            return undefined;
        }
        const raw = this.base ? this.raw + Path.separator : this.raw;
==>    if (!path.raw.startsWith(raw)) {
            return undefined;
        }
        const relativePath = path.raw.substr(raw.length);
        return new Path(relativePath);
    }
```
I agree that I shouldn't modify URI and just use Path inside of [plugin-icon-theme-service.ts](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/main/browser/plugin-icon-theme-service.ts). But I also think the change in [path.ts](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/path.ts) is good, because the normalization should make them equal... A normalized `C:/...` Path should equal a normalized `/c:/...` Path. My new changes dispense the change of [uri.ts](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/uri.ts) but still include the `Path` fix. It is easy to fix it inside of [path.ts](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/path.ts) without having to add long checks because they're already there in `Path.ts`.

Reply:
What is the value of `packagePath` for you?

Reply:
The troubles come from the fact that we violate https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#uri-over-path we should rather replace `PluginModel.packagePath` with `PluginModel.packageUri` if we want to use it on the frontend and backend. cc @benoitf  

path.ts code is correctly aligned with [vscode-uri](https://github.com/microsoft/vscode-uri/blob/master/src/index.ts#L1084) logic and cannot be changed

Reply:
Ok, I'll have a look. That was my 2nd thought to change what's coming in and no try to force it into the right shape. Will update this PR today

Reply:
We could add `packageUri` next `packagePath` for now and deprecate the latter without changing other clients for it in this PR. Another PR can clean up replacement of `packagePath` by `packageUri`.

Reply:
At the moment, I've set it up that the old `packageUri` that's now called `packageUriFromPath` has priority over the new `packageUri` if it is defined. If not, the new `packageUri` steps in and everything should work. The checks are still necessary for me though, because it wouldn't work without the front `/` and the `\` being converted to `/` The changes are only made on windows though and shouldn't affect any other operating system.